kernel/src/main/java/org/kframework/utils/TimingCollector.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2014-2021 K Team. All Rights Reserved.
+package org.kframework.utils;
+
+import scala.Tuple2;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+@ThreadSafe
+public class TimingCollector {
+    final Deque<Tuple2<String, String>> messages = new ConcurrentLinkedDeque<>();
```

Comment:
We need to check if there is any chance this will block or not? @radumereuta please link us to the documentation.